examples/java/src/test/java/org/apache/beam/examples/complete/TrafficMaxLaneFlowIT.java

```diff
@@ -63,14 +72,28 @@ public void testE2ETrafficMaxLaneFlow() throws Exception {
     this.options.setBigQueryDataset(this.outputDatasetId);
     this.options.setBigQueryTable(this.outputTable);
     TrafficMaxLaneFlow.runTrafficMaxLaneFlow(this.options);
+    FluentBackoff backoffFactory =
+        FluentBackoff.DEFAULT.withMaxRetries(4).withInitialBackoff(Duration.standardSeconds(1L));
+    Sleeper sleeper = Sleeper.DEFAULT;
+    BackOff backoff = BackOffAdapter.toGcpBackOff(backoffFactory.backoff());
+    String res = "empty_result";
+    int attemptNum = 1;
+    do {
+      QueryResponse response =
+          this.bqClient.queryWithRetries(
+              String.format(
+                  "SELECT count(*) as total FROM [%s:%s.%s]",
+                  this.projectId, this.outputDatasetId, this.outputTable),
+              this.projectId);
 
-    QueryResponse response =
-        this.bqClient.queryWithRetries(
-            String.format(
-                "SELECT count(*) as total FROM [%s:%s.%s]",
-                this.projectId, this.outputDatasetId, this.outputTable),
-            this.projectId);
-    String res = response.getRows().get(0).getF().get(0).getV().toString();
+      try {
+        res = response.getRows().get(0).getF().get(0).getV().toString();
+        break;
+      } catch (Exception e) {
```

Comment:
Shall we limit catch to NullPointerException?

Reply:
The overall idea is we want to retry certain times to get ride of any Exception caused by BQ not by our service. I would prefer using Exception. Instead, how about also logging stacktrace of exception?

Reply:
I think their are 2 sort of cases that needs retry.
1. Error in getting response from BQ (which is not happening in the try-catch block.
2. Early result causing null pointer. (happening in the try-catch block) 
As for logging , I think we should log stack trace after we fail the last retry.